thrifty-schema/src/main/java/com/microsoft/thrifty/schema/Constant.java

```diff
@@ -258,6 +258,12 @@ public void validate(Linker linker, ThriftType expected, ConstValueElement value
                 } else {
                     throw new IllegalStateException("bad enum literal: " + value.value());
                 }
+            } else if (named instanceof Typedef) {
```

Comment:
Usually we use `type.getTrueType()` to resolve typedefs; we just need to do that to the type being validated.

I think the bug is actually on line 79.  When we look up a validator based on its type, the first thing that happens is that we resolve the typedef.  When we then call `validate`, we pass the original, unresolved, type.  So line 79 ought to say:

```java
Validators.forType(expectedType).validate(linker, expectedType.getTrueType(), value);
```

or, split it up:

```java
expectedType = expectedType.getTrueType();
Validators.forType(expectedType).validate(linker, expectedType, value);
```

This way, when we later look up a `Named` instance for the given type, we will get back the `EnumType` and not the `Typedef`.